source/lib/core/rocpd/data_processor.cpp

```diff
@@ -0,0 +1,672 @@
+            .set_values(pid, _upid, nid, ppid, pid, init, fini, start, end, command,
+                        environment, extdata)
+            .get_query_string());
+}
+
+size_t
+data_processor::insert_agent(size_t node_id, size_t pid, const char* agent_type,
+                             size_t absolute_index, size_t logical_index,
+                             size_t type_index, uint64_t uuid, const char* name,
+                             const char* model_name, const char* vendor_name,
+                             const char* product_name, const char* user_name,
+                             const char* extdata)
+{
+    data_storage::queries::table_insert_query query;
+    data_storage::database::get_instance().execute_query(
+        query.set_table_name("rocpd_info_agent_" + _upid)
+            .set_columns("guid", "nid", "pid", "type", "absolute_index", "logical_index",
+                         "type_index", "uuid", "name", "model_name", "vendor_name",
+                         "product_name", "user_name", "extdata")
+            .set_values(_upid, node_id, pid, agent_type, absolute_index, logical_index,
+                        type_index, uuid, name, model_name, vendor_name, product_name,
+                        user_name, extdata)
+            .get_query_string());
+
+    return data_storage::database::get_instance().get_last_insert_id();
+}
+
+void
+data_processor::insert_track(const char* track_name, size_t node_id, size_t process_id,
+                             std::optional<size_t> thread_id, const char* extdata)
+{
+    if(_tracks.find(track_name) != _tracks.end())
+    {
+        ROCPROFSYS_WARNING(2, "Fail to add track %s, already exist!\n", track_name);
+        return;
+    }
+
+    auto name_id = insert_string(track_name);
+
+    data_storage::queries::table_insert_query query;
+    data_storage::database::get_instance().execute_query(
+        query.set_table_name("rocpd_track_" + _upid)
+            .set_columns("guid", "nid", "pid", "tid", "name_id", "extdata")
+            .set_values(_upid, node_id, process_id, thread_id, name_id, extdata)
+            .get_query_string());
+
+    auto track_id       = data_storage::database::get_instance().get_last_insert_id();
+    _tracks[track_name] = track_name_map{ track_id, name_id };
+}
+
+void
+data_processor::insert_pmc_description(
+    size_t node_id, size_t process_id, size_t agent_id, const char* target_arch,
+    size_t event_code, size_t instance_id, const char* name, const char* symbol,
+    const char* description, const char* long_description, const char* component,
+    const char* units, const char* value_type, const char* block, const char* expression,
+    uint32_t is_constant, uint32_t is_derived, const char* extdata)
+{
+    auto it = _pmc_descriptor_map.find({ agent_id, name });
+    if(it != _pmc_descriptor_map.end())
+    {
+        ROCPROFSYS_WARNING(
+            0, "Insert PMC description failed! Error: PMC descriptor already exist!\n");
```

Comment:
```suggestion
            0, "Insert PMC description failed! Error: PMC descriptor (name:%s) (ID:%lu) already exist!\n",
            name, agent_id);
```
Adding a little more detail to the warning message.